src/DotNetProjectFile.Analyzers/Analyzers/MsBuild/IndentXml.cs

```diff
@@ -0,0 +1,63 @@
+﻿using Microsoft.CodeAnalysis.Text;
+
+namespace DotNetProjectFile.Analyzers.MsBuild;
+
+[DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+public sealed class IndentXml(char ch, int repeat) : MsBuildProjectFileAnalyzer(Rule.IndentXml)
+{
+    private readonly char Char = ch;
+    private readonly int Repeat = repeat;
+
+    public IndentXml() : this(' ', 2) { }
+
+    protected override void Register(ProjectFileAnalysisContext context)
+        => Walk(context.Project, context.Project, context.Project.Text, context);
+
+    private void Walk(MsBuildProject project, Node node, SourceText text, ProjectFileAnalysisContext context)
+    {
+        Report(project, node, text, context, true);
+        Report(project, node, text, context, false);
+
+        foreach (var child in node.Children)
+        {
+            Walk(project, child, text, context);
+        }
+    }
+
+    private void Report(MsBuildProject project, Node node, SourceText text, ProjectFileAnalysisContext context, bool start)
+    {
+        if (CheckSelfClosingEnd() || StartAndEndOnSameLine()) { return; }
+
+        var element = start ? node.Positions.StartElement : node.Positions.EndElement;
+
+        if (!ProperlyIndented(element))
+        {
+            var name = start ? node.LocalName : '/' + node.LocalName;
+            context.ReportDiagnostic(Descriptor, project, element, name);
+        }
+
+        bool CheckSelfClosingEnd() => !start && node.Positions.IsSelfClosing;
```

Comment:
These 2 could just be variables, since their value will never change during the execution. I think it would make it easier to read through the method.